core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -105,14 +94,14 @@ public void delete(TableIdentifier tableIdentifier, Table table, RemovalCause ca
   }
 
   private Cache<TableIdentifier, Table> createTableCache(Ticker ticker) {
-    Caffeine<TableIdentifier, Table> cacheBuilder = Caffeine
+    Caffeine<Object, Object> cacheBuilder = Caffeine
```

Comment:
Is there a reason we need the generic types here?

Reply:
Yes there is. We need it because caffeine has no way of determining the type based only in `Caffeine.newBuilder().softValues()`. Previously we specified the removalListener here, so it could get the specific types

Reply:
So technically it works if you put the types, but intellij does kind of freak out (even if you put the types on the right side and the left).

This is normal, to use `<Object, Object>` until something gives it type definitions (after which there are strict rules about using that chained fluent builder because it stores type info from the responseListener, which doesn't come until later now).

Starting with `<Object, Object>` is very normal with Caffeine's builder.